Microsoft.Packaging.Tools/tasks/Trimming/NuGetPackageNode.cs

```diff
@@ -12,15 +12,13 @@ namespace Microsoft.DotNet.Build.Tasks
     internal sealed class NuGetPackageNode : IIsIncluded
     {
         private const string NuGetPackageDependencies = "dependencies";
-        private List<NuGetPackageNode> _dependencies = new List<NuGetPackageNode>();
-        private IEnumerable<string> _dependencyIds;
+        private HashSet<NuGetPackageNode> _dependencies = new HashSet<NuGetPackageNode>();
```

Comment:
What do you think about NuGetPackageNode overriding Equals/GetHashCode so this HashSet works correctly?